json-endpoints/class.wpcom-json-api-get-site-endpoint.php

```diff
@@ -227,44 +227,44 @@ public function build_current_site_response() {
 			array_intersect( $default_fields, $this->fields_to_include ) :
 			$default_fields;
 
-		if ( ! $this->has_blog_access( $this->api->token_details, $blog_id ) ) {
+		$has_blog_access = $this->has_blog_access( $this->api->token_details );
+		$has_user_access = $this->has_user_access();
+
+		if ( ! $has_user_access && ! $has_blog_access ) {
+			// Users without user or blog access, only return `$no_member_fields`.
 			$response_keys = array_intersect( $response_keys, self::$no_member_fields );
+		} elseif ( $has_user_access && ! current_user_can( 'edit_posts' ) ) {
+			// Subscriber level user, don't return site options.
+			$response_keys = array_diff( $response_keys, [ 'options' ] );
 		}
 
 		return $this->render_response_keys( $response_keys );
 	}
 
 	/**
-	 * Checks that the current user has access to the current blog,
-	 * and failing that checks that we have a valid blog token.
+	 * Checks that the current user has access to the current blog.
 	 *
-	 * @param $token_details array Details obtained from the authorization token
-	 * @param $blog_id int The server-side blog id on wordpress.com
+	 * @return bool Whether or not the current user can access the current blog.
+	 */
+	private function has_user_access() {
+		return is_user_member_of_blog( get_current_user_id(), get_current_blog_id() );
+	}
+
+	/**
+	 * Checks if the request has a valid blog token for the current blog.
 	 *
+	 * @param array $token_details Access token for the api request.
 	 * @return bool
 	 */
-	private function has_blog_access( $token_details, $blog_id ) {
-		$current_blog_id = (  defined( 'IS_WPCOM' ) && IS_WPCOM ) ?
-			$blog_id :
-			get_current_blog_id();
```

Comment:
As best I can tell, this code is no different from using `get_current_blog_id()` on it's own.

For `WPCOM`, the $blog_id passed into the function is from `$GLOBALS['blog_id']`. For Jetpack, `current_blog_id()` is used in the conditional. So I removed this code.

Reply:
That's a great question, thanks for digging up that part of the history. From the discussion in D7311-code, it appears the concern was with Jetpack multisite installs. I'll spin one up and test--and add a code comment if that's the case.

Reply:
I've tested this against a subdirectory site on a multisite network using the same protocol
- Requests to `/sites/{site}` and `/sites/{site}?force=wpcom` through public-api.wordpress.com
- Requests as public, subscriber, and admin user on the site, also requests `as_blog` after `switch_to_blog` in `wp shell`

I can't see any problems with the responses. The correct sub directory site title and information looks to be returned, the responses match what I see in production. Can you think of another way to call this endpoint on a multisite installation that I might be missing?

Otherwise it seems that something has improved with the api code such that the `IS_WPCOM` specific logic is no longer needed for Jetpack sites on a multisite network. The blog_id seems to be working correctly using `get_current_blog_id()`, as far as I can tell.